// See README.txt for information and build instructions.
//
// Note: START and END tags are used in comments to define sections used in
// tutorials.  They are not part of the syntax for Protocol Buffers.
//
// To get an in-depth walkthrough of this file and the related examples, see:
// https://developers.google.com/protocol-buffers/docs/tutorials

// [START declaration]
syntax = "proto3";
package posting_message;

//import "google/protobuf/timestamp.proto";
// [END declaration]

// [START messages]
// Message for add Doc
message doc_info {
  int32 id = 1;
  int32 term_frequency = 2;
  repeated int32 positions = 3;  // vector of positions
} 

// Posting
message Offset {
  int32 start_offset = 1;
  int32 end_offset = 2;
}

message Posting {
  int32 docID = 1;
  int32 term_frequency = 2;
  repeated Offset positions = 3;  // vector of positions
}

// Posting with precomputed scores fro snippet generating
message Passage_Score {
  int32 passage_id = 1;
  float score = 2;
}
message Passage_Split {
  int32 start_offset = 1;
  int32 len = 2;
}

message Posting_Precomputed_4_Snippets {
  int32 docID = 1;
  int32 term_frequency = 2;
  repeated Offset offsets = 3;  // vector of positions
  repeated Passage_Score passage_scores = 4;
  map<int32, Passage_Split> passage_splits = 5;
}

// Posting List

// Our address book file is just one of these.
message PostingList {
  int32 num_postings = 1;        // number of postings in this list
  repeated Posting postings = 2;        // posting list when creating index
}
// [END messages]
